lib/view_component/test_helpers.rb

```diff
@@ -225,6 +225,11 @@ def vc_test_request
         end
     end
 
+    def before_setup
+      @request = __vc_render_preview_controller.request
```

Comment:
This isn't correct since this is creating the request from the preview controller, not using the same request the `request` method uses. Can we update this to use the proper request?

Reply:
I'm not sure I follow. On line 78  we have `previews_controller = __vc_render_preview_controller`, `#request` tracks to   `ViewComponent::Base#request` which memoizes the request.

If you debug the call you can see that the request objects are indeed the same.

`@request ->  #<ActionDispatch::TestRequest:0x000000010c675838>`
`preview_controller.request -> #<ActionDispatch::TestRequest:0x000000010c675838>`

These are identical. Am I missing something?

Reply:
```suggestion
      @request = vc_test_request
```

That wasn't completely clear in hindsight, sorry about that. 

We shouldn't use `__vc_render_preview_controller.request` here because we don't need to couple the `@request` to the preview controller or how it's construction. Also not every test requires a preview controller to be present either, so we should skip instantiating one except where necessary.

Reply:
Actually we do. The Warden context is not correctly set up in `ActionDispatch::TestRequest.create`. I tested this myself with one  of your earlier suggestions.

https://github.com/ViewComponent/view_component/pull/1849#discussion_r1323312240

Reply:
We don't always use the preview controller instance to render since users can pass their own controller class to us and `render_inline` uses its own controller.

What does Devise need the controller instance for?

Reply:
https://github.com/heartcombo/devise/blob/f6e73e5b5c8f519f4be29ac9069c6ed8a2343ce4/lib/devise/test/controller_helpers.rb#L44

Reply:
@BlakeWilliams to be clear, this path is ONLY triggered if you call `current_user` in any of your components. Which is kindof edge case ish right now. I actually refactored so that current_user is only ever passed into components when they need it.

That said, I read somewhere that the long term goal of ViewComponent is that it would be used for page rendering as well, which will make this a lot less of an edge case.